libs/features/webapp-prompt/webapp-prompt.js

```diff
@@ -35,6 +35,13 @@ const getIcon = (content) => {
   return icons.company;
 };
 
+const geoRoutingActive = () => !!document.querySelector('.locale-modal-v2')
+          || !!document.querySelector('.locale-modal');
+
+const waitForClosedGRMThen = (loadPEP) => {
+  window.addEventListener('milo:modal:closed', loadPEP);
```

Comment:
sorry for doing this, but I just realized that it's possible for two dialogs to be open.
see this link - https://stage--cc--adobecom.hlx.page/products/photoshop/ai?milolibs=pep-georouting&skipPepEntitlements=true#langnav
we need to add a check here that geoRoutingActive() is now false.

Reply:
For some reason the `#langnav` short circuits my check and the pep dialog shows up anyway. Even while the GRM is not yet dismissed.

Reply:
Also since the #langnav shows up late, it's now an open question whether the pep modal should show up anyway or wait until it's gone.

Reply:
Also, I've discovered that when the event is emitted it won't necessarily trigger PEP's `init` method, since it's possible for the DOM to still exist at that point. I'm looking into why this might be.

Reply:
completely agree. 
In my opinion PEP should not be appearing if any modal is open, but since that makes your task more complicated I didn't mention it.  
Maybe you can mention that to your PdM and if they deem it necessary then tackle that in a separate task.

Reply:
oh and btw, you want to clean up your listener once you've triggered it once. 
https://dev.to/js_bits_bill/addeventlistener-once-js-bits-565d

Reply:
@sharmrj it seems to me that the right solution would be to change the modal.js so that the close event is sent after removing the modal.  
@JasonHowellSlavin can you weigh in if you see issues with moving the window.dispatchEvent of milo:modal:closed after removing the modal?

Reply:
Hi @vhargrave, I do think both you and @sharmrj have identified a key issue with the `milo:modal:closed` dispatch being placed too high in the function and not reflecting what is actually occurring. I don't see an issue with moving it down, but since there are a few other files that rely on it, it might be good to raise that as an issue and specifically test for those implementations. 

It also looks like @sharmrj has done some experimentation and found the performance impact of the current code to be negligible.

Reply:
@JasonHowellSlavin ok, thanks for confirming 👍